@orbeckst 

Here are the outputs for the benchmarks running locally : 

EinsteinMSDCustom on a log scale
<img width="1728" height="994" alt="Screenshot 2026-03-05 at 19 22 09" src="https://github.com/user-attachments/assets/ae34af41-4046-4771-a6e2-a1c82e26cf7a" />

EinsteinMSDExample on a linear scale
<img width="1728" height="994" alt="Screenshot 2026-03-05 at 19 21 29" src="https://github.com/user-attachments/assets/50a15ce1-e4dd-4513-9f70-d187286021d4" />

I ran these benchmarks over the last 50 commits of the develop branch with a step size of 5.
I initially tried to run it over the last 150 commits with a step size of 10, but after `develop~60` there was an error about C code compilation. I have pasted a part of the error here for reference.

```text
MDAnalysis/lib/src/transformations/transformations.c:778:53: error: use of undeclared identifier 'NPY_IN_ARRAY'
 778 |     *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_IN_ARRAY);
     |                                                     ^
MDAnalysis/lib/src/transformations/transformations.c:778:53: error: use of undeclared identifier 'NPY_IN_ARRAY'
MDAnalysis/lib/src/transformations/transformations.c:778:53: error: use of undeclared identifier 'NPY_IN_ARRAY'
MDAnalysis/lib/src/transformations/transformations.c:794:57: error: use of undeclared identifier 'NPY_ALIGNED'; did you mean 'NPY_VALID'?
 794 |         *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_ALIGNED);
     |                                                         ^~~~~~~~~~~
     |                                                         NPY_VALID
/private/var/folders/fn/k_8ycp0d0fxcnrsp3jdqw80h0000gn/T/build-env-wy87mqfy/lib/python3.11/site-packages/numpy/_core/include/numpy/ndarrayobject.h:91:28: note: expanded from macro 'PyArray_FROM_OTF'
  91 |                         (((flags) & NPY_ARRAY_ENSURECOPY) ? \
     |                            ^
/private/var/folders/fn/k_8ycp0d0fxcnrsp3jdqw80h0000gn/T/build-env-wy87mqfy/lib/python3.11/site-packages/numpy/_core/include/numpy/ndarrayobject.h:92:28: note: expanded from macro 'PyArray_FROM_OTF'
  92 |         NPY_VALID=0,
     |         ^
MDAnalysis/lib/src/transformations/transformations.c:811:57: error: use of undeclared identifier 'NPY_IN_ARRAY'
MDAnalysis/lib/src/transformations/transformations.c:849:33: error: use of undeclared identifier 'NPY_ENSURECOPY'                                       ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
```
I believe this is an environment configuration issue, rather than an issue with the benchmark itself. Please correct me if I'm wrong.

--- a/package/AUTHORS
+++ b/package/AUTHORS
@@ -271,9 +271,9 @@ Chronological list of authors
   - Mohammad Ayaan
   - Khushi Phougat
   - Kushagar Garg
-  - Ayush Agarwal
   - Jeremy M. G. Leung
   - Harshit Gajjela
+  - Ayush Agarwal
 
 External code
 -------------

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -15,8 +15,8 @@ The rules for this file:
 
 -------------------------------------------------------------------------------
 ??/??/?? IAlibay, orbeckst, marinegor, tylerjereddy, ljwoods2, marinegor,
-         spyke7, talagayev, tanii1125, BradyAJohnston, hejamu, Ayush-Agarwal-G1THUB,
-         jeremyleung521, harshitgajjela-droid
+         spyke7, talagayev, tanii1125, BradyAJohnston, hejamu,
+         jeremyleung521, harshitgajjela-droid, aygarwal
 
  * 2.11.0
 
@@ -41,11 +41,11 @@ Fixes
    DSSP by porting upstream PyDSSP 0.9.1 fix (Issue #4913)
 
 Enhancements
+ * Added benchmarks for package.MDAnalysis.analysis.msd.EinsteinMSD (PR #5277)
  * Improved performance of `AtomGroup.wrap()` with compounds (PR #5220)
  * Adds support for parsing `.tpr` files produced by GROMACS 2026.0
  * Enables parallelization for analysis.diffusionmap.DistanceMatrix
    (Issue #4679, PR #4745)
- * Added benchmarks for package.MDAnalysis.analysis.msd.EinsteinMSD (PR #5277)
 
 Changes
  * The msd.py inside analysis is changed, and ProgressBar is implemented inside

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -15,8 +15,8 @@ The rules for this file:
 
 -------------------------------------------------------------------------------
 ??/??/?? IAlibay, orbeckst, marinegor, tylerjereddy, ljwoods2, marinegor,
-         spyke7, talagayev, tanii1125, BradyAJohnston, hejamu,
-         jeremyleung521, harshitgajjela-droid, aygarwal
+         spyke7, talagayev, tanii1125, BradyAJohnston, hejamu, jeremyleung521,
+         harshitgajjela-droid, aygarwal
 
  * 2.11.0
 

